x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -431,6 +437,14 @@ private void handleLoadSuccess(String modelId,
                     // If it isn't, we flip an `isLoaded` flag, and increment the model counter to make sure if it is evicted
                     // between now and when the listeners access it, the circuit breaker reflects actual usage.
                     localModelCache.computeIfAbsent(modelId, modelAndConsumerLoader);
+                    // We should start tracking on successful load. It will stop being tracked once it evacuates the cache and is no
+                    // longer a referenced model
+                    // NOTE: It is not possible to change the referenced models without locking on `loadingListeners`
+                    //       So, if the model is evacuated from cache immediately after checking that it was present,
+                    //       the feature usage will still be tracked.
+                    if (License.OperationMode.BASIC.equals(trainedModelConfig.getLicenseLevel()) == false) {
```

Comment:
I thought about putting the license tracking within the `LocalModel` object itself, but this posed a couple of problems. 

 - What if a LocalModel object is created but not cached (starts tracking)
 - The same model is then loaded into cache via another process (starts tracking)
 - LocalModel object that was not previously cached is destroyed (stops tracking)
 - Now the cached object which refers to the same model is no longer tracking as the uniqueness is handled by the `model_id`

I am just not 100% sure how to handle this weird scenario so I think putting the tracking here in the loading service is good for now. 

I think once we move java models over to TrainedModelAllocations, these execution paths will become clearer.